utilities/pulumi_state_splitter/tests/test_cli.py

```diff
@@ -0,0 +1,128 @@
+        result = runner.invoke(
+            pulumi_state_splitter.cli,
+            args,
+            catch_exceptions=False,
+        )
+        self.assertEqual(result.exit_code, want_exit_code, result.output)
+
+    @parameterized.parameterized.expand(
+        itertools.product(
+            [False, True],
+            [
+                "split",
+                "unsplit",
+            ],
+            [
+                True,
+                False,
+            ],
+            [
+                [
+                    "test-project-1/test-stack-1",
+                    "test-project-1/test-stack-2",
+                ],
+                [],
+            ],
+        )
+    )
+    def test_split_unsplit(self, set_backend_dir, command, all_stacks, stacks):
+        """Testing `pulumi_state_splitter.cli`, split and unsplit commands."""
+
+        input_, want = (
+            (data.multi_stack_unsplit(), data.multi_stack_split())
+            if command == "split"
+            else (data.multi_stack_split(), data.multi_stack_unsplit())
+        )
+
+        args = []
+        if set_backend_dir:
+            args.extend(
+                [
+                    "--backend-directory",
+                    self._tmp_dir,
+                ]
+            )
+        args.append(command)
+        for stack in stacks:
+            args.extend(["--stack", stack])
+        if all_stacks:
+            args.append("--all-stacks")
+
+        if all_stacks == bool(stacks):
+            with self.assertRaises(ValueError):
+                self._cli_run(args)
+            return
+
+        if stacks:
+            # if stacks is specified, then test-project-2 is not being processed
+            if command == "split":
+                want[".pulumi"] = {
+                    "stacks": {
+                        "test-project-2": input_[".pulumi"]["stacks"]["test-project-2"]
+                    }
+                }
+                want.pop("test-project-2")
+            else:
+                want[".pulumi"]["stacks"].pop("test-project-2")
+                want["test-project-2"] = input_["test-project-2"]
+
+        input_.save(self._tmp_dir)
+
+        with contextlib.chdir("." if set_backend_dir else self._tmp_dir):
+            self._cli_run(args)
+
+        got = util.Directory.load(self._tmp_dir)
+        want.compare(got, self)
+
+    def test_split_run(self):
+        """Testing `pulumi_state_splitter.cli`, successful run command"""
+        input_ = data.multi_stack_split()
+        input_.save(self._tmp_dir)
+        with contextlib.chdir(self._tmp_dir):
+            self._cli_run(
+                [
+                    "run",
+                    "--all-stacks",
+                    "--",
+                    "cp",
+                    "-r",
+                    ".pulumi/stacks/test-project-2/test-stack-3.json",
+                    ".pulumi/stacks/test-project-2/test-stack-4.json",
+                ]
+            )
+        want = input_
```

Comment:
I guess you wanted to run some assertions here instead of plain assignments.

Reply:
Thanks, fixed in 6ad6d00a1809635e202910980b261e5f56ac827f .